### PR TITLE
refactor: improve config view to be more visually appealing

### PR DESCRIPTION
**Pull request by nikolaischunk**
I rewrote the current config view code to better align with the "old" / previous design we used!

| Old View | New View |
| --------------- | --------------- | 
| Image description           | Image description |

### CODE REVIEW

File: .github/hooks/pre-commit [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-yarn lint-staged
```
Comment (nikolaischunk) [inline .github/hooks/pre-commit:3]: @Swiizyy do we still need this even? not sure

File: src/commands/Admin/config.ts [hunk 2]
```diff
@@ -146,32 +147,39 @@ export class ConfigCommand extends BirthdayySubcommand {
 	) {
 		settings ??= {};
 		const t = await fetchT(interaction);
+		if (interaction.guild === null) throw new Error('Guild not found');
```
Comment (nikolaischunk) [inline src/commands/Admin/config.ts:150]: this should never happen in theory!

Comment (Swiizyy) [reply src/commands/Admin/config.ts:150]: It will never happen you can delete it

File: src/config.ts [hunk 3]
```diff
@@ -280,3 +280,6 @@ function parseWebhookLog(): WebhookClientData | null {
 }
 
 export const WEBHOOK_LOG = parseWebhookLog();
+
+export const DEFAULT_ANNOUNCEMENT_MESSAGE =
+	envParseString('DEFAULT_ANNOUNCEMENT_MESSAGE') || 'Default Announcement Message';
```
Comment (Swiizyy) [inline src/config.ts:285]:
```suggestion
	envParseString('DEFAULT_ANNOUNCEMENT_MESSAGE', 'Default Announcement Message');
```

**Review by swiizyy — CHANGES_REQUESTED** [review]

File: src/commands/Admin/config.ts (hunk 2 above)
Comment (nikolaischunk) [reply src/commands/Admin/config.ts:150]: it is done that we dont have to use ? after interaction.guild later.
image

**Comment by Swiizyy** [comment]
@nikolaischunk Do you still have any changes to make?

File: src/lib/util/environment.ts [hunk 4]
```diff
@@ -12,15 +11,18 @@ export const DEBUG = envParseBoolean('DEBUG', true);
 export const APP_ENV = envParseString('APP_ENV');
 
 // GENERIC
-export const { CLIENT_NAME, BOT_AVATAR } = process.env;
+export const CLIENT_NAME = envParseString('CLIENT_NAME', 'Birthdayy');
+export const CLIENT_AVATAR = envParseString(
+	'CLIENT_AVATAR',
+	'https://cdn.discordapp.com/avatars/1039089174948626473/8d7f1e8a8b9f6e2b2f4b9a2c9b2b3a7e.webp'
```
Comment (Swiizyy) [inline src/lib/util/environment.ts:17]: This image is empty is this the expected result?

**Comment by Swiizyy** [comment]
Capture d’écran 2024-05-31 à 12 20 34
Wouldn't it be better like that?

**Comment by nikolaischunk** [comment]
It looks super good, I agree! Well done :)

File: src/lib/util/environment.ts (hunk 4 above)
Comment (nikolaischunk) [reply src/lib/util/environment.ts:17]: my bad, we should use the bot avatar via code not hardcoded, it would be better I think

Comment (Swiizyy) [reply src/lib/util/environment.ts:17]: Yes, I think it would be better 👍

**Pull request merged**